lib/src/default_revset_engine.rs

```diff
@@ -644,11 +644,7 @@ impl<'index, 'heads> EvaluationContext<'index, 'heads> {
                 predicate: build_predicate_fn(self.store.clone(), self.index, predicate),
             })),
             RevsetExpression::AsFilter(candidates) => self.evaluate(candidates),
-            RevsetExpression::Present(candidates) => match self.evaluate(candidates) {
-                Ok(set) => Ok(set),
-                Err(RevsetError::NoSuchRevision(_)) => Ok(Box::new(EagerRevset::empty())),
-                r @ Err(RevsetError::AmbiguousIdPrefix(_) | RevsetError::StoreError(_)) => r,
-            },
+            RevsetExpression::Present(candidates) => self.evaluate(candidates),
```

Comment:
This change reminds me that we talked about having something like `expect_one()` and `expect_any()`, which would behave similar to `present()` but would be handled at evaluation time, I suppose.

(No action needed.)

Reply:
Yes, it'll be an evaluation error if the expect/actual counts differ.